stm/stm.ml

```diff
@@ -3008,7 +3008,7 @@ let process_transaction ~doc ?(newtip=Stateid.fresh ())
               let bname = VCS.mk_branch_name x in
               let opacity_of_produced_term = function
                 (* This AST is ambiguous, hence we check it dynamically *)
-                | VernacInstance (false, _,_ , None, _) -> GuaranteesOpacity
+                | VernacInstance (_,_ , None, _) -> GuaranteesOpacity (* XXX really? can't we close it by Defined? *)
```

Comment:
What does this mean?

Reply:
Well, it probably means I don't understand the classification, but why would `Instance foo : C.` "guarantee opacity"?

Reply:
IIRC GuaranteesOpacity means that if the proof is closed by Qed it's really opaque. Derive breaks this. I don't know why Instance would though.

Reply:
Aaaah! Thanks for the clarification, I'll remove the comment before this is merge, then.